src/plugins/data/public/query/query_string/dataset_service/lib/index_type.ts

```diff
@@ -120,11 +121,10 @@ const fetchDataSources = async (client: SavedObjectsClientContract) => {
     perPage: 10000,
   });
   const dataSources: DataStructure[] = response.savedObjects
-    .filter(
-      (savedObject) =>
-        typeof savedObject.attributes?.dataSourceEngineType === 'string' &&
-        !savedObject.attributes?.dataSourceEngineType?.includes('OpenSearch Serverless')
-    )
+    .filter((savedObject) => {
+      const coercedVersion = semver.coerce(savedObject.attributes.dataSourceVersion);
```

Comment:
This attribute always exists?

Reply:
yeah @Flyingliuhub @ZilongX confirming this is the correct attribute?

We should get cypress tests once i merge the workflow in https://github.com/opensearch-project/OpenSearch-Dashboards/pull/8703

Reply:
yes `dataSourceVersion` would always exist, it may carry empty value for invalid datasource but the filed itself would be there